pyNastran/bdf/cards/material_deps.py

```diff
@@ -259,6 +259,192 @@ def write_card(self, size: int=8, is_double: bool=False) -> str:
             return self.comment + print_card_8(card)
         return self.comment + print_card_16(card)
 
+class MATDMG(MaterialDependence):
+    """
+    Material Properties for Progressive Ply Failure
+
+    Defines material properties and parameters for progressive ply failure in composite solid elements defined with the
+    PCOMPS bulk entry. Used in combination with MAT11 entries that have the same MID. Valid for SOLs 401 and 402.
+
+    +--------+-------+--------+-------+------+-----+-----+-----+-----+
+    |   1    |   2   |    3   |  4    |  5   |  6  |  7  |  8  |  9  |
+    +========+=======+========+=======+======+=====+=====+=====+=====+
+    | MATDMG |  MID  | PPFMOD |       |      |     |     |     |     |
+    +--------+-------+--------+-------+------+-----+-----+-----+-----+
+    |        | COEF1 | COEF2  | -etc- |      |     |     |     |     |
+    +--------+-------+--------+-------+------+-----+-----+-----+-----+
+
+    Format for PPFMOD == "UD"
+    +--------+--------+------+------+------+---------+---------+------+------+
+    |   1    |   2    |   3  |  4   |  5   |    6    |    7    |  8   |  9   |
+    +========+========+======+======+======+=========+=========+======+======+
+    | MATDMG |  MID   |  UD  |      |      |         |         |      |      |
+    +--------+--------+------+------+------+---------+---------+------+------+
+    |        |  Y012  | YC12 | YS12 | YS22 | Y11LIMT | Y11LIMC | KSIT | KSIC |
+    +--------+--------+------+------+------+---------+---------+------+------+
+    |        |   B2   |  B3  |  A   | LITK |   BIGK  |   EXPN  | TAU | ADEL  |
+    +--------+--------+------+------+------+---------+---------+------+------+
+    |        | PLYUNI | TID  | HBAR | DMAX |   PE    |         |     |       |
+    +--------+--------+------+------+------+---------+---------+------+------+
+
+    Format for PPFMOD == "EUD"
+    +--------+--------+------+------+-------+---------+---------+------+------+
+    |   1    |   2    |   3  |  4   |  5    |    6    |    7    |  8   |  9   |
+    +========+========+======+======+=======+=========+=========+======+======+
+    | MATDMG |   MID  | EUD  |      |       |         |         |      |      |
+    +--------+--------+------+------+-------+---------+---------+------+------+
+    |        |  Y012  | YC12 |   K  | ALPHA | Y11LIMT | Y11LIMC | KSIT | KSIC |
+    +--------+--------+------+------+-------+---------+---------+------+------+
+    |        |   B2   |  B3  |   A  | LITK  |  BIGK   |  EXPN   | TAU  | ADEL |
+    +--------+--------+------+------+-------+---------+---------+------+------+
+    |        |  USER  | R01  | HBAR | DMAX  |   DS    |   GIC   | GIIC | GIIIC|
+    +--------+--------+------+------+-------+---------+---------+------+------+
+
+    """
+    type = 'MATDMG'
+
+    def __init__(self, mid, ppf_model,
```

Comment:
Fields that are optional (e.g., anything with *or_blank like double_or_blank) should have a default.  The default should be consistent with the MATDMG.add_card method 

Same goes for the add_matdmg method in the add_card.py file